File: p04_functions/exercises/perfect_number.py
def is_perfect(num):
    divisors = []
    for n in range(1, num):
        if num % n == 0:
            divisors.append(n)
    if not sum(divisors) == num:
        return False
    return True
number = int(input())
if is_perfect(number):
    print("We have a perfect number!")
else:
    print("It's not so perfect.")

File: SoftUni-JS-Front-End/functionsAndStatementsExercise/perfectNumber.js
function solve(number) {
    let total = 0
    for (let i = 0; i < number; ++i) {
        if (number % i === 0) {
            total += i
        }
    }
    if (total === number) {
        return "We have a perfect number!"
    }else {
        return "It's not so perfect."
    }
}

console.log((solve(1236498)))